src/main/java/javaslang/collection/List.java

```diff
@@ -133,12 +133,12 @@
             @SuppressWarnings("unchecked")
             final List<T> list = (List<T>) elements;
             return list;
-        } else if (elements instanceof ArrayList) {
+        } else if (elements instanceof ArrayList || elements instanceof Vector) {
             @SuppressWarnings("unchecked")
-            final ArrayList<T> arrayList = (ArrayList<T>) elements;
+            final java.util.List<T> indexedList = (java.util.List<T>) elements;
```

Comment:
@szarnekow That helps! Didn't know that.

Do you want to create a pull request or should I...?